### ISSUE TITLE
Issue 006: 初期 Coarse experiments YAML を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#291]
# Issue 006: 初期 Coarse experiments YAML を追加する

## 背景

Issue 001/002 で experiment registry と runner/config generator wiring を追加し、Issue 003-005 で site54-local Coarse runtime extension を追加する。

この issue では、比較の最初の対象として以下の experiment YAML を追加する。

```text
baseline
velocity_mask_default
trend_prior_default
velocity_mask_trend_prior_default
```

既存パラメータを default として、baseline は空 overlay のまま維持する。非 baseline は `site54_oof_experiment.coarse_train_runtime` のみを追加し、fold list や通常の training hyperparameter は変えない。

## 重要な制約

- `packages/seisai-engine/src/**` は変更しない。
- `packages/seisai-pick/src/**` は変更しない。
- `baseline.yaml` は overlay 空のままにする。
- 既存 `make_coarse_fold_configs.py` の baseline training parameter は変更しない。
- 初期 experiments は Coarse train runtime extension のみを対象にする。heldout coarse inference, physics, fine train/infer はこの issue では変えない。
- fold list、train / inner_valid / heldout split は変更しない。

## 現在の repo 前提

repo 既存の site54 physical band は以下に定義されている。

```text
proc/fbpick/site54/oof/scripts/make_physics_fold_configs.py
```

```yaml
physical_band:
  vmin_m_s: 300.0
  vmax_m_s: 6000.0
  t0_lo_ms: -20.0
  t0_hi_ms: 200.0
```

`velocity_mask_default` では、この既存 band を velocity mask の default として使う。

`trend_prior_default` では、既存 `TrendPriorConfig` / `IRLSStrategy` の default に寄せつつ、Coarse 用に `fb_idx_key: fb_idx_coarse` を指定する。

## 追加・変更対象

追加または更新:

```text
proc/fbpick/site54/oof/experiments/README.md
proc/fbpick/site54/oof/experiments/baseline.yaml
proc/fbpick/site54/oof/experiments/velocity_mask_default.yaml
proc/fbpick/site54/oof/experiments/trend_prior_default.yaml
proc/fbpick/site54/oof/experiments/velocity_mask_trend_prior_default.yaml
```

必要なら tests を追加。

変更禁止:

```text
packages/seisai-engine/src/**
packages/seisai-pick/src/**
```

## 1. `baseline.yaml`

baseline は既存設定そのものを表す。overlay は空。

```yaml
schema_version: 1
id: baseline
title: Existing site54 OOF baseline
description: >-
  Existing site54 OOF settings without Coarse runtime extensions.
base: null
tags:
  - baseline
config_overrides:
  coarse_train: {}
  coarse_train_smoke: {}
  coarse_infer: {}
  physics: {}
  physics_qc: {}
  fine_train: {}
  fine_train_smoke: {}
  fine_infer: {}
metadata:
  preserves_existing_defaults: true
```

## 2. `velocity_mask_default.yaml`

Coarse train の base KL loss 前に velocity mask を適用する。

```yaml
schema_version: 1
id: velocity_mask_default
title: Coarse train velocity-mask default
description: >-
  Apply the existing site54 physical velocity band as a differentiable mask on
  Coarse training logits before the base KL loss.
base: baseline
tags:
  - coarse
  - velocity_mask
config_overrides:
  coarse_train:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: true
          apply_train: true
          apply_inner_valid: false
          vmin: 300.0
          vmax: 6000.0
          t0_lo_ms: -20.0
          t0_hi_ms: 200.0
          taper_ms: 8.0
          power: 1.0
          offsets_key: offsets
          dt_key: dt_sec
          channels: 0
          ensure_nonempty: true
          aux_key: velocity_mask
        trend_prior_loss:
          enabled: false
  coarse_train_smoke:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: true
          apply_train: true
          apply_inner_valid: false
          vmin: 300.0
          vmax: 6000.0
          t0_lo_ms: -20.0
          t0_hi_ms: 200.0
          taper_ms: 8.0
          power: 1.0
          offsets_key: offsets
          dt_key: dt_sec
          channels: 0
          ensure_nonempty: true
          aux_key: velocity_mask
        trend_prior_loss:
          enabled: false
  coarse_infer: {}
metadata:
  default_source:
    velocity_band: proc/fbpick/site54/oof/scripts/make_physics_fold_configs.py:physical_band
```

## 3. `trend_prior_default.yaml`

Coarse train loss に trend prior CE を追加する。

```yaml
schema_version: 1
id: trend_prior_default
title: Coarse train trend-prior default
description: >-
  Add a differentiable trend-prior CE regularizer to Coarse training loss.
base: baseline
tags:
  - coarse
  - trend_prior
config_overrides:
  coarse_train:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: false
        trend_prior_loss:
          enabled: true
          weight: 0.03
          fit_kind: irls
          section_len: 128
          stride: 64
          huber_c: 1.345
          iters: 3
          vmin_m_s: 300.0
          vmax_m_s: 6000.0
          prior_sigma_ms: 20.0
          prior_alpha: 1.0
          prior_conf_gate: 0.5
          prior_log_eps: 1.0e-4
          conf_floor: 0.2
          conf_power: 0.5
          logit_clip: 30.0
          tau: 1.0
          channels: 0
          offsets_key: offsets
          fb_idx_key: fb_idx_coarse
          dt_key: dt_sec
          aux_key: trend_prior
  coarse_train_smoke:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: false
        trend_prior_loss:
          enabled: true
          weight: 0.03
          fit_kind: irls
          section_len: 128
          stride: 64
          huber_c: 1.345
          iters: 3
          vmin_m_s: 300.0
          vmax_m_s: 6000.0
          prior_sigma_ms: 20.0
          prior_alpha: 1.0
          prior_conf_gate: 0.5
          prior_log_eps: 1.0e-4
          conf_floor: 0.2
          conf_power: 0.5
          logit_clip: 30.0
          tau: 1.0
          channels: 0
          offsets_key: offsets
          fb_idx_key: fb_idx_coarse
          dt_key: dt_sec
          aux_key: trend_prior
  coarse_infer: {}
metadata:
  default_source:
    trend_defaults: packages/seisai-engine/src/seisai_engine/postprocess/trend_prior_op.py:TrendPriorConfig
    fit_defaults: packages/seisai-pick/src/seisai_pick/trend/trend_fit_strategy.py:IRLSStrategy
```

## 4. `velocity_mask_trend_prior_default.yaml`

velocity mask と trend prior を同時に有効化する。順序は以下を想定する。

```text
model logits -> velocity_mask -> base KL loss
                              -> trend_prior_loss
```

つまり trend prior も velocity-masked logits を見る。

```yaml
schema_version: 1
id: velocity_mask_trend_prior_default
title: Coarse train velocity-mask + trend-prior default
description: >-
  Apply velocity masking and add trend-prior CE in Coarse training.
base: baseline
tags:
  - coarse
  - velocity_mask
  - trend_prior
config_overrides:
  coarse_train:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: true
          apply_train: true
          apply_inner_valid: false
          vmin: 300.0
          vmax: 6000.0
          t0_lo_ms: -20.0
          t0_hi_ms: 200.0
          taper_ms: 8.0
          power: 1.0
          offsets_key: offsets
          dt_key: dt_sec
          channels: 0
          ensure_nonempty: true
          aux_key: velocity_mask
        trend_prior_loss:
          enabled: true
          weight: 0.03
          fit_kind: irls
          section_len: 128
          stride: 64
          huber_c: 1.345
          iters: 3
          vmin_m_s: 300.0
          vmax_m_s: 6000.0
          prior_sigma_ms: 20.0
          prior_alpha: 1.0
          prior_conf_gate: 0.5
          prior_log_eps: 1.0e-4
          conf_floor: 0.2
          conf_power: 0.5
          logit_clip: 30.0
          tau: 1.0
          channels: 0
          offsets_key: offsets
          fb_idx_key: fb_idx_coarse
          dt_key: dt_sec
          aux_key: trend_prior
  coarse_train_smoke:
    site54_oof_experiment:
      coarse_train_runtime:
        enabled: true
        velocity_mask:
          enabled: true
          apply_train: true
          apply_inner_valid: false
          vmin: 300.0
          vmax: 6000.0
          t0_lo_ms: -20.0
          t0_hi_ms: 200.0
          taper_ms: 8.0
          power: 1.0
          offsets_key: offsets
          dt_key: dt_sec
          channels: 0
          ensure_nonempty: true
          aux_key: velocity_mask
        trend_prior_loss:
          enabled: true
          weight: 0.03
          fit_kind: irls
          section_len: 128
          stride: 64
          huber_c: 1.345
          iters: 3
          vmin_m_s: 300.0
          vmax_m_s: 6000.0
          prior_sigma_ms: 20.0
          prior_alpha: 1.0
          prior_conf_gate: 0.5
          prior_log_eps: 1.0e-4
          conf_floor: 0.2
          conf_power: 0.5
          logit_clip: 30.0
          tau: 1.0
          channels: 0
          offsets_key: offsets
          fb_idx_key: fb_idx_coarse
          dt_key: dt_sec
          aux_key: trend_prior
  coarse_infer: {}
metadata:
  order: velocity_mask_then_base_and_trend_prior
```

## README 更新

`proc/fbpick/site54/oof/experiments/README.md` に以下を含める。

- 追加済み experiment 一覧。
- 各 experiment の意図。
- 実行例。

```bash
OOF_ROOT=/workspace/proc/fbpick/site54/oof
RUN_ID=coarse_ablation_v1

python $OOF_ROOT/scripts/run_site54_oof_cv.py \
  --run-id $RUN_ID \
  --experiment baseline \
  --stage prepare_configs

python $OOF_ROOT/scripts/run_site54_oof_cv.py \
  --run-id $RUN_ID \
  --experiment velocity_mask_trend_prior_default \
  --stage prepare_configs
```

## 受け入れ条件

- `packages/seisai-engine/src/**` と `packages/seisai-pick/src/**` に変更がない。
- 4つの YAML が `experiment_registry.py --list` に表示される。
- 4つの YAML が `experiment_registry.py --experiment <id> --dump` で resolve できる。
- `baseline` の `config_overrides` は空。
- 非 baseline は `coarse_train` と `coarse_train_smoke` に同じ runtime extension を入れる。
- `coarse_infer` overlay は空。heldout inference の挙動はこの issue では変えない。
- `velocity_mask_default` は site54 既存 `physical_band` と同じ速度帯を使う。
- `trend_prior_default` は `fb_idx_key: fb_idx_coarse` を使う。

## 確認コマンド

```bash
cd /workspace

python proc/fbpick/site54/oof/scripts/experiment_registry.py \
  --cv-root /workspace/proc/fbpick/site54/oof \
  --list

for exp in baseline velocity_mask_default trend_prior_default velocity_mask_trend_prior_default; do
  python proc/fbpick/site54/oof/scripts/experiment_registry.py \
    --cv-root /workspace/proc/fbpick/site54/oof \
    --experiment "$exp" \
    --dump >/tmp/${exp}.resolved.yaml
done

python proc/fbpick/site54/oof/scripts/make_coarse_fold_configs.py \
  --cv-root /workspace/proc/fbpick/site54/oof \
  --run-id coarse_exp_yaml_probe \
  --experiment velocity_mask_trend_prior_default

python - <<'PY'
from pathlib import Path
import yaml
p = Path('/workspace/proc/fbpick/site54/oof/runs/coarse_exp_yaml_probe/experiments/velocity_mask_trend_prior_default/configs/fold00/01_coarse_train.yaml')
if not p.exists():
    # If Issue 002 chose a different explicit run_root in local test, adjust manually.
    raise SystemExit(f'missing expected config: {p}')
cfg = yaml.safe_load(p.read_text())
runtime = cfg['site54_oof_experiment']['coarse_train_runtime']
assert runtime['velocity_mask']['enabled'] is True
assert runtime['trend_prior_loss']['enabled'] is True
assert runtime['trend_prior_loss']['fb_idx_key'] == 'fb_idx_coarse'
print('ok')
PY
```
